unleash-yggdrasil/src/lib.rs

```diff
@@ -529,7 +530,9 @@ impl EngineState {
             self.count_toggle(name, enabled);
 
             if enabled {
-                self.check_variant_by_toggle(toggle, context)
+                let mut variant = self.check_variant_by_toggle(toggle, context).unwrap_or_default();
+                variant.feature_enabled = true;
```

Comment:
I don't think this is quite going to work for us here. My feeling is that check_variant_by_toggle should be told what the state of the parent toggle is rather than setting a potentially invalid default and hoping the caller fixes it.

From a quick glance, it looks like everywhere that `check_variant_by_toggle` is called, we have enough information to pass that in. Then it would be much tougher to make a mistake on this

Can we change the function signature to `check_variant_by_toggle(toggle, context, is_enabled)`?